### PR TITLE
Add documentation for using rerank and normalization processors together

### LINKED ISSUE #7392: [DOC] Change documentation for Rerank processor, add special case of Normalization processor
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
Existing documentation for Rerank processor should be enhanced, we need to add a special case when Rerank and Normalization processors are working together. In such case scores of Normalization processor will be updated by Rerank, and that may be an unexpected behavior for end user.

This behavior is not related to any specific version and was there from the moment of release of both processors. Ideally this change should be added as soon as possible, this behavior is there for users of 2.12 and above.

Suggested change:

**Special Scenario: Interaction with Normalization Processor and Hybrid Query**
When using the rerank processor in conjunction with the normalization processor and a hybrid query, there is an important behavior to be aware of: the rerank processor will alter the result scores of the documents. This is due to the fact that the rerank processor operates after the normalization processor in the search pipeline.

**Understanding the Order of Processing**
Normalization Processor: This processor normalizes the scores of the documents based on the configured normalization method. For more details on how this processor works, refer to the [Normalization Processor documentation](https://opensearch.org/docs/latest/search-plugins/search-pipelines/normalization-processor/).
Rerank Processor: Following the normalization, the rerank processor will further adjust the scores of the documents. This adjustment can significantly impact the final ordering of search results.
**Implications**
- **Score Modification:** The rerank processor will modify the scores that were initially adjusted by the normalization processor, potentially leading to different ranking results than initially expected.
- **Hybrid Queries:** In the context of hybrid queries, where multiple types of queries and scoring mechanisms are combined, this behavior is particularly noteworthy. The combined scores from the initial query will be normalized first and then reranked, resulting in a two-stage scoring modification. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Rerank processor page: https://opensearch.org/docs/latest/search-plugins/search-relevance/reranking-search-results/
Normalization processor page: https://opensearch.org/docs/latest/search-plugins/search-pipelines/normalization-processor/

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Closes #7392  

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@martin-gaievski Could you please review this PR?

**Review by martin-gaievski — APPROVED** [review]
Looks good to me, thank you!

File: _search-plugins/search-relevance/reranking-search-results.md [hunk 1]
```diff
@@ -115,4 +115,19 @@ POST /my-index/_search
...
+
+## Using rerank and normalization processors together
+
+When you use a rerank processor in conjunction with a [normalization processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/) and a hybrid query, the rerank processor alters the final document scores. This is because the rerank processor operates after the normalization processor in the search pipeline.
+{: .note}
+
+The processing order is as follows: 
+
+- Normalization processor: This processor normalizes the document scores based on the configured normalization method. For more information, see [Normalization processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/).
+- Rerank processor: Following the normalization, the rerank processor further adjusts the document scores. This adjustment can significantly impact the final ordering of search results.
```
Comment (natebower) [inline _search-plugins/search-relevance/reranking-search-results.md:128]:
```suggestion
- Rerank processor: Following normalization, the rerank processor further adjusts the document scores. This adjustment can significantly impact the final ordering of search results.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just one minor deletion. Thanks!

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**